src/Store.h

```diff
@@ -191,10 +193,7 @@ class StoreEntry : public hash_link, public Packable
     bool hasMemStore() const { return mem_obj && mem_obj->memCache.index >= 0; }
 
     /// whether this entry can feed collapsed requests and only them
-    bool hittingRequiresCollapsing() const { return EBIT_TEST(flags, ENTRY_REQUIRES_COLLAPSING); }
-
-    /// allow or forbid collapsed requests feeding
-    void setCollapsingRequirement(const bool required);
+    bool hittingRequiresCollapsing() const { return mem_obj && (mem_obj->getReply()->pstate < Http::Message::psParsed); }
```

Comment:
I know you have you checked that mem_obj is not deleted for entries that should have true hittingRequiresCollapsing(). What about entries that are created by CF initiators in non-SMP setups? Do they always get mem_obj before the first hittingRequiresCollapsing() call? If yes, can we adjust the code so that we can assert() true hittingRequiresCollapsing() just before Store::Controller::allowCollapsing() returns true?

Reply:
> Do they always get mem_obj before the first hittingRequiresCollapsing() call? 

Yes, since Controller::allowCollapsing() is always called just after storeCreateEntry()/storeCreatePureEntry(), creating the memory object. I added this assert.